extra/helpers.py

```diff
@@ -26,32 +24,28 @@ def early_exec(x):
   return early_exec
 
 def proc(itermaker, q) -> None:
-  for x in itermaker(): q.put(x)
-  q.put(None)
-  q.close()
+  try:
+    for x in itermaker(): q.put(x)
+  except Exception as e:
+    q.put(e)
+  finally:
+    q.put(None)
+    q.close()
 
 class _CloudpickleFunctionWrapper:
-  def __init__(self, fn):
-    self.fn = fn
-
-  def __getstate__(self):
-    return cloudpickle.dumps(self.fn)
-
-  def __setstate__(self, pfn):
-    self.fn = cloudpickle.loads(pfn)
-
-  def __call__(self, *args, **kwargs) -> Any:
-    return self.fn(*args, **kwargs)
+  def __init__(self, fn): self.fn = fn
+  def __getstate__(self): return cloudpickle.dumps(self.fn)
+  def __setstate__(self, pfn): self.fn = cloudpickle.loads(pfn)
+  def __call__(self, *args, **kwargs) -> Any:  return self.fn(*args, **kwargs)
 
 def cross_process(itermaker, maxsize=16):
   q: multiprocessing.Queue = multiprocessing.Queue(maxsize)
   # multiprocessing uses pickle which cannot dump lambdas, so use cloudpickle.
   p = multiprocessing.Process(target=proc, args=(_CloudpickleFunctionWrapper(itermaker), q))
-  #p.daemon = True
+  p.daemon = True
   p.start()
-
-  # TODO: write tests and handle exit case
```

Comment:
I figured the graceful exit case was already handled, i.e. when `None` is returned, but the case of an exception was not.

Reply:
Yea this is an improvement